packages/core-transaction-pool/lib/pool-wallet-manager.js

```diff
@@ -63,125 +63,91 @@ module.exports = class PoolWalletManager extends WalletManager {
+      errors.push(
+        `Can't apply transaction ${
+          transaction.id
         }: delegate name already taken.`,
       )
-
-      // NOTE: We use the vote public key, because vote transactions
-      // have the same sender and recipient.
+      // NOTE: We use the vote public key, because vote transactions have the same sender and recipient.
     } else if (
       type === TRANSACTION_TYPES.VOTE &&
       !database.walletManager.__isDelegate(asset.votes[0].slice(1))
     ) {
       logger.error(
         `[PoolWalletManager] Can't apply vote transaction: delegate ${
           asset.votes[0]
-        } does not exist.`,
-        JSON.stringify(data),
+        } does not exist. Data: ${JSON.stringify(transaction)}`,
       )
 
-      throw new Error(
-        `[PoolWalletManager] Can't apply transaction ${data.id}: delegate ${
+      errors.push(
+        `Can't apply transaction ${transaction.id}: delegate ${
           asset.votes[0]
         } does not exist.`,
       )
-    } else if (this.__isException(data)) {
+    } else if (this.__isException(transaction)) {
       logger.warn(
-        'Transaction forcibly applied because it has been added as an exception:',
-        data,
+        `Transaction forcibly applied because it has been added as an exception: ${transaction}`,
       )
-    } else if (!sender.canApply(data, errors)) {
+    } else if (!sender.canApply(transaction, errors)) {
       const message = `[PoolWalletManager] Can't apply transaction id:${
-        data.id
-      } from sender:${sender.address} due to ${JSON.stringify(errors)}`
-      logger.error(message)
-      logger.debug(
-        `[PoolWalletManager] Audit: ${JSON.stringify(
-          sender.auditApply(data),
-          null,
-          2,
-        )}`,
-      )
-
-      throw new Error(message)
+        transaction.id
+      } from sender:${sender.address}`
+      logger.error(`${message} due to ${JSON.stringify(errors)}`)
+      errors.unshift(message)
     }
 
-    sender.applyTransactionToSender(data)
-
-    if (type === TRANSACTION_TYPES.TRANSFER) {
-      const recipient = this.findByAddress(recipientId)
-      recipient.applyTransactionToRecipient(data)
-    }
-
-    return transaction
+    return errors.length === 0
   }
 
   /**
-   * Apply the given block to a delegate in the pool wallet manager.
-   * We apply only the block reward and fees, as transaction are already be applied
-   * when entering the pool. Applying only if delegate wallet is in pool wallet manager
-   * @param {block}
+   * Remove the given transaction from a a sender only.
+   * @param  {Transaction} transaction
+   * @return {Transaction}
    */
-  applyPoolBlock(block) {
-    // if delegate in poll wallet manager - apply rewards
-    if (this.exists(block.data.generatorPublicKey)) {
-      const delegateWallet = this.findByPublicKey(block.data.generatorPublicKey)
-      delegateWallet.applyBlock(block.data)
-    }
-  }
+  revertTransaction(transaction) {
```

Comment:
"revertTransaction" is misleading if we are going to revert only the sender balance.